### ISSUE TITLE
There is wrong about running the command nvidia-smi

### ISSUE THREAD

**Issue by 251099155** [jcjohnson/neural-style#335]
My system is ubuntu 14.04 on win10.I have installed the CUDA 8.0 ,but when i input the command nvidia-smi,there is an error like this:

modprobe: ERROR: ../libkmod/libkmod.c:556 kmod_search_moddep() could not open moddep file '/lib/modules/3.4.0+/modules.dep.bin'
modprobe: ERROR: ../libkmod/libkmod.c:556 kmod_search_moddep() could not open moddep file '/lib/modules/3.4.0+/modules.dep.bin'
modprobe: ERROR: ../libkmod/libkmod-module.c:809 kmod_module_insert_module() could not find module by name='nvidia_367'
modprobe: ERROR: could not insert 'nvidia_367': Function not implemented
NVIDIA-SMI has failed because it couldn't communicate with the NVIDIA driver. Make sure that the latest NVIDIA driver is installed and running.

how to solve it?

**Comment by htoyryla** [comment]
The installation of nvidia driver has failed. From this it is impossible to know why. One reason could be missing kernel headers. NVIDIA documentation and forum are best sources for help for this kind of problem.

**Comment by 251099155** [comment]
@htoyryla I have try to install the Navidia Driver，but it say that can't find kernel 3.4.0+.I use the command apt-cache search linux-headers,but it seems have no the headers file of linux-headers-3.4.0+
![qq 20161007172149](https://cloud.githubusercontent.com/assets/13912267/19185088/a5ed3fba-8cb2-11e6-999e-7ccc40b89c7e.png)
![image](https://cloud.githubusercontent.com/assets/13912267/19185276/7aa2892c-8cb3-11e6-8d06-878a235d38ff.png)

**Comment by htoyryla** [comment]
Where did you get this 3.4.0+ from? It does not sound right. Which kernel do you have? What does uname -r give?

The installation guide at http://docs.nvidia.com/cuda/cuda-installation-guide-linux/index.html#axzz4MOGxcY2j gives the following supported versions. 

Ubuntu 16.04	4.4.0
Ubuntu 14.04	3.13	

I am running 14.04 with a 4.4.0 kernel (although still cuda 8.0rc, have not yet moved to 8.0).

**Comment by 251099155** [comment]
@htoyryla Here is some info of my system.So what is the problem?
![image](https://cloud.githubusercontent.com/assets/13912267/19186627/c37ce4ca-8cb9-11e6-88f4-a0e55237527b.png)

**Comment by htoyryla** [comment]
Can't really help anymore. You seem to have a 3.4.0 kernel but instead of a precise version number, it has this +. Google doesn't either give anything helpful on "kernel 3.4.0+".

Anyhow, 3.4.0 is old already and might not even work with cuda 8.0 (the docs state 3.13 and 4.4.0).

**Comment by htoyryla** [comment]
Just realized (based on another issue) that you are running on Windows bash? Others, too, seem to complaining of not finding the kernel header for 3.4.0. I wonder whether it is possible to update the kernel in Windows bash?

**Comment by 251099155** [comment]
yes,I'm runnning on Win10's bash.Thank u a lot.but I don't konw how to change my system's kernel..

**Comment by hanyaqian** [comment]
@251099155 hello, i meet the same problem, if you had solved it?

**Comment by Matrixsun** [comment]
I meet a similarity problem.  my system is redhat7 ,my GPU is K80 in Azure .my cuda is cuda8. my error is 
NVIDIA-SMI has failed because it couldn't communicate with the NVIDIA driver. Make sure that the latest NVIDIA driver is installed and running.

solve it by 
 modprobe nvidia

then every thing is OK . my Gold.

**Comment by rizky** [comment]
@Matrixsun I have the same issue, but when I do modprobe nvidia I received this error: "modprobe: ERROR: could not insert 'nvidia_375': No such device".

Would you mind elaborate more on how do you solve "NVIDIA-SMI has failed because it couldn't communicate with the NVIDIA driver"

my setting, Ubuntu 14.04, K80, in Azure

**Comment by Matrixsun** [comment]
@rizkyario  sorry for my simple.  Details as follows:
Azure's Linux system lacks a lot of basic packages. My solution is one by one to install those packages I need before the installation of cuda. 
1、install GCC
   yum install gcc*  
2、install dkms
   wget http://dl.fedoraproject.org/pub/epel/7/x86_64/e/epel-release-7-9.noarch.rpm
   rpm -ivh epel-release-7-9.noarch.rpm
   yum install --enablerepo=epel dkms 
3、Install kernel related components
    yum install kernel*
4、install cuda
5、modprobe nvidia
then it is  OK .

**Comment by Laqshay** [comment]
I am trying to run NVIDIA driver on AWS running with Ubuntu 16.04. I have installed kernel headers and successfully installed the NVIDIA-375.26 driver that comes along with official CUDA-8.0 release. However I keep getting "NVIDIA-SMI has failed because it couldn't communicate with the NVIDIA driver. Make sure that the latest NVIDIA driver is installed and running", no matter how many times I reinstall, run `sudo update-initramfs -u` and reboot. When I tried the method recommended by @Matrixsun, I got the following error: "modprobe: FATAL: Module nvidia not found in directory /lib/modules/4.4.0-1020-aws". What do I do to resolve this?

**Comment by 3DTOPO** [comment]
> I am trying to run NVIDIA driver on AWS running with Ubuntu 16.04

What instance type?

**Comment by Laqshay** [comment]
p2.xlarge with NVIDIA Tesla K80
